uberenv.py

```diff
@@ -978,9 +978,9 @@ def concretize_spack_env(self):
         sexe(spack_concretize_cmd, echo=True)
 
     def clean_build(self):
-        # clean out any spack cached stuff (except build stages, downloads, &
-        # spack's bootstrapping software)
-        cln_cmd = "{0} clean --misc-cache --failures --python-cache".format(self.spack_exe(use_spack_env=False))
+        # clean out any spack cached stuff (except build stages, downloads)
+        # note: we now clean out spack bootstrap b/c ~/.spack contents can undermine bootstrap
+        cln_cmd = "{0} clean -b --misc-cache --failures --python-cache".format(self.spack_exe(use_spack_env=False))
```

Comment:
```suggestion
        cln_cmd = "{0} clean --bootstrap --misc-cache --failures --python-cache".format(self.spack_exe(use_spack_env=False))
```
nitpick for consistency and clarity